### ISSUE TITLE
Log where a configuration error occurs

### ISSUE THREAD

**Issue by TheVastyDeep** [elastic/logstash#14243]
The compiler [raises an exception](https://github.com/elastic/logstash/blob/0605b75420f9723e13ab694b01cb98efdc3d196d/logstash-core/lib/logstash/compiler.rb#L32) when there is a syntax error in the configuration. I realize that a lot of code passes "null" for source_with_metadata, but if SourceWithMetadata.protocol is "file" then SourceWithMetadata.id should be the path. Including that id in the error message would make it much easier to find problems in the configuration when it is available.

**Comment by steevhise** [comment]
agreed.  it's a constant problem, especially because the configuration files seem to be so touchy about syntax, white space, etc etc.
and in my searches for answers to these problems, i've seen many many others with similar problems that could have been easily fixed if just they knew where in the config file the problem is.  dozens of old issues on ES discuss where people are like "getting mystery error" and then someone answering "oh that means you have a problem in your config file" and then poster says "i can't find where" and then a month later "oh i finally found it, needed brackets around the value" or something dumb.  
To summarize: this is **really** something that would help a **lot** of people use Logstash.

**Comment by fransf-wtax** [comment]
Has this really not been addressed yet? I'm getting a mystery `"Expected one of [ \\t\\r\\n], \"#\", \"input\", \"filter\", \"output\" at line 26, column 1 (byte 869) after "` error. I have 12 config files that are all included from `pipelines.xml`. Now I have to open each one of them one by one, go to line 26, and see if anything looks off.

At the end of the day, I don't think Ruby syntax is a great choice for a config file. Ruby is code, not data. Supporting something more friendly like YAML would be the way to go, IMHO.
